### PR TITLE
Change locks to System.Threading.Lock

### LINKED ISSUE #11672: Refactor `lock` on `object` usage to `System.Threading.Lock`
In .NET9 there is the new `System.Threading.Lock`
https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock?view=net-9.0

> It is recommended to use the [EnterScope](https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock.enterscope?view=net-9.0) method with a language construct that automatically disposes the returned [Lock.Scope](https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock.scope?view=net-9.0) such as the C# using keyword, or to use the C# lock keyword, as these ensure that the lock is exited in exceptional cases. These patterns might also have performance benefits over using Enter/TryEnter and Exit. The following code fragment illustrates various patterns for entering and exiting a lock.

Refactor the `object` used for the lock to `Lock` where appropriate. The new `Lock` class when used with `lock()` will not use `system.threading.monitor`, but instead will use `system.threading.lock`. 

Benchmarks in this blog post suggest a 25% performance improvement in lock operations:
https://steven-giesel.com/blogPost/4cf6d48a-ec9d-4c68-961c-31fd8d8c1340

There are some limitations, I think the team is best positioned to decide whats best here.

**We won't make changes until C# takes the change to the `lock` keyword out of preview.**
https://github.com/dotnet/csharplang/issues/7104

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Change internal lock objects to System.Threading.Lock

Fixes #11672
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11841)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Timer.cs [hunk 1]
```diff
@@ -133,23 +137,25 @@ public int Interval
...
+                    return;
+                }
+
+                _interval = value;
+                if (Enabled)
                 {
-                    _interval = value;
-                    if (Enabled)
+                    // Change the timer value, don't tear down the timer itself.
+                    if (!DesignMode && _timerWindow is not null)
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Timer.cs:156]: Consider combining these

**Review by lonitra — APPROVED** [review]
LGTM :shipit: Minor comment, but can be addressed in the next.

File: src/System.Windows.Forms/src/System/Windows/Forms/Timer.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Timer.cs:156]: Will do in a follow up.

**Comment by MarkCiliaVincenti** [comment]
I'm intrigued by this implementation. Does this not harm performance for < .NET 9.0? Have you benchmarked?

**Pull request merged**